lib/core/src/lib/testing/unit-testing-utils.ts

```diff
@@ -0,0 +1,465 @@
+
+    static doubleClickByDataAutomationId(debugElement: DebugElement, dataAutomationId: string): void {
+        const element = UnitTestingUtils.getByDataAutomationId(debugElement, dataAutomationId);
+        element.triggerEventHandler('dblclick', new MouseEvent('dblclick'));
+    }
+
+    static blurByCSS(debugElement: DebugElement, selector: string): void {
+        const element = UnitTestingUtils.getByCSS(debugElement, selector);
+        element.triggerEventHandler('blur', new FocusEvent('blur'));
+    }
+
+    static hoverOverByCSS(debugElement: DebugElement, selector: string): void {
+        const element = UnitTestingUtils.getByCSS(debugElement, selector);
+        element.triggerEventHandler('mouseenter', new MouseEvent('mouseenter'));
+    }
+
+    static hoverOverByDataAutomationId(debugElement: DebugElement, dataAutomationId: string): void {
+        const element = UnitTestingUtils.getByDataAutomationId(debugElement, dataAutomationId);
+        element.triggerEventHandler('mouseenter', new MouseEvent('mouseenter'));
+    }
+
+    static mouseLeaveByCSS(debugElement: DebugElement, selector: string): void {
+        const element = UnitTestingUtils.getByCSS(debugElement, selector);
+        element.triggerEventHandler('mouseleave', new MouseEvent('mouseleave'));
+    }
+
+    static mouseLeaveByDataAutomationId(debugElement: DebugElement, dataAutomationId: string): void {
+        const element = UnitTestingUtils.getByDataAutomationId(debugElement, dataAutomationId);
+        element.triggerEventHandler('mouseleave', new MouseEvent('mouseleave'));
+    }
+
+    static keyBoardEventByCSS(debugElement: DebugElement, selector: string, event: 'keyup' | 'keydown', code: string, key: string): void {
+        const element = UnitTestingUtils.getByCSS(debugElement, selector);
+        element.nativeElement.dispatchEvent(new KeyboardEvent(event, { code, key }));
+    }
+
+    static dispatchCustomEventByCSS(debugElement: DebugElement, selector: string, eventName: string): void {
+        const element = UnitTestingUtils.getByCSS(debugElement, selector);
+        element.nativeElement.dispatchEvent(new CustomEvent(eventName));
+    }
+
+    /** Input related methods */
+
+    static getInputByCSS(debugElement: DebugElement, selector: string): HTMLInputElement {
+        return UnitTestingUtils.getByCSS(debugElement, selector)?.nativeElement;
+    }
+
+    static getInputByDataAutomationId(debugElement: DebugElement, dataAutomationId: string): HTMLInputElement {
+        return UnitTestingUtils.getByDataAutomationId(debugElement, dataAutomationId)?.nativeElement;
+    }
+
+    static fillInputByCSS(debugElement: DebugElement, selector: string, value: any): void {
+        const input = UnitTestingUtils.getInputByCSS(debugElement, selector);
+        input.value = value;
+        input.dispatchEvent(new Event('input'));
+    }
+
+    static fillInputByDataAutomationId(debugElement: DebugElement, dataAutomationId: string, value: any): void {
```

Comment:
similar situation like above - we can set value to string because value of input has string type